pyiceberg/table/__init__.py

```diff
@@ -538,6 +566,74 @@ def overwrite(
                 for data_file in data_files:
                     update_snapshot.append_data_file(data_file)
 
+    def delete(self, delete_filter: Union[str, BooleanExpression], snapshot_properties: Dict[str, str] = EMPTY_DICT) -> None:
+        if (
```

Comment:
Shall we add docstring for this new public method?

Reply:
Ahh, missed that one, thanks!